integration_test/chrome/starting_sessions_test.exs

```diff
@@ -0,0 +1,87 @@
+defmodule Wallaby.Integration.Chrome.StartingSessionsTest do
+  use ExUnit.Case, async: false
+
+  import Wallaby.SettingsTestHelpers
+  import Wallaby.TestSupport.ApplicationControl
+  import Wallaby.TestSupport.TestScriptUtils
+  import Wallaby.TestSupport.TestWorkspace
+
+  alias Wallaby.Experimental.Chrome
+  alias Wallaby.TestSupport.Chrome.ChromeTestScript
+
+  @moduletag :capture_log
+
+  setup [:stop_wallaby, :create_test_workspace]
+
+  test "works when chromedriver starts immediately", %{workspace_path: workspace_path} do
+    {:ok, chromedriver_path} = Chrome.find_chromedriver_executable()
+
+    test_script_path =
+      chromedriver_path
+      |> ChromeTestScript.build_wrapper_script()
+      |> write_test_script!(workspace_path)
+
+    ensure_setting_is_reset(:wallaby, :chromedriver)
+    Application.put_env(:wallaby, :chromedriver, path: test_script_path)
+
+    assert :ok = Application.start(:wallaby)
```

Comment:
The `flunk` in `stop_wallaby` is in an `on_exit` callback, so the flunk will occur only if it can't restart wallaby after the test is done. 

Since wallaby reads its config and starts worker pools on Application startup, in order to change settings for the driver we need to:

1. Stop wallaby
2. Adjust the environment for the new script path/settings
3. Start wallaby back up

In the future, I think it would be really nice to allow more overrides in `Wallaby.start_session` (like which driver to use, whether or not to start a chromedriver instance or use a remote instance, etc), instead of relying so heavily on the application config that we're having to start and stop the full application during tests and can only use one driver (phantom, selenium, chromedriver) at a time. That's definitely outside the scope of this PR and should be discussed more, but I figured I just should share what I've been thinking for the longer term.

Reply:
Is Wallaby not already started at the beginning of the test since it's restarted in the `on_exit` callback? Or does calling `Application.start/1` restart the application? (I was thinking it nooped if it was already started)

Reply:
I stopped wallaby before the test is run on [line 14](https://github.com/elixir-wallaby/wallaby/pull/512/files#diff-e1a21673c207293787a531827a25c722R14) in the setup block:

```elixir
setup [:stop_wallaby, :create_test_workspace]
```

I figured that way I can start the tests by configuring wallaby before I start it back up (and so I don't forget to stop it and have my config changes not matter). Now that I look at it, I do see that setup step could be easy to miss, though. Is there another way I should write this that would be more clear?

Reply:
🤦‍♂ 

I realized that the `on_exit` through me off, in reality, the steps are

1. stop wallaby
1. reset env stuff
1. start wallaby
1. test
1. stop wallaby
1. start wallaby

I might split it up into two functions. One stops wallaby `stop_wallaby` and the other defines the `on_exit` and is named `restart_on_exit!` or something like that.

Reply:
I split it into 2 functions in the setup block: `stop_wallaby` and `restart_wallaby_on_exit!`. Do you prefer `Application.stop(:wallaby)` as the first line in each test instead of having a function that's called during setup?